### PR TITLE
(Feature) Gender verification. Issue basic identity certificate

### PR DESCRIPTION
**Pull request by johnsmith-gooddollar**
# Description

- scaffolding 
- small refactoring 
- aws client 
- aws  gender + age utils method 
- endpoint 
- faceted age check method
- unit tests

About #453 #454 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Comment by johnsmith-gooddollar** [comment]
only unit tests were left, so could be reviewed now

File: src/server/verification/processor/EnrollmentProcessor.js [hunk 1]
```diff
@@ -292,8 +342,8 @@ const enrollmentProcessors = new WeakMap()
 
 export default (storage, log) => {
   if (!enrollmentProcessors.has(storage)) {
-    log = log || logger.child({ from: 'EnrollmentProcessor' })
-    const enrollmentProcessor = new EnrollmentProcessor(Config, storage, AdminWallet, log)
+    const processorLogger = log || logger.child({ from: 'EnrollmentProcessor' })
```
Comment (johnsmith-gooddollar) [inline src/server/verification/processor/EnrollmentProcessor.js:345]: actually processor shouldn't encapsulate logger. it should be passed as optional param to each method.
this should be fixed in a separate PR.

File: src/server/verification/processor/EnrollmentProcessor.js [hunk 2]
```diff
@@ -140,6 +145,46 @@ class EnrollmentProcessor {
...
+      FV_IDENTIFIER_MSG2({ account: toChecksumAddress(gdAddress) }),
+      ''
+    )
+
+    if (!strcasecmp(signer, gdAddress)) {
+      throw new Error(`identifier signer doesn't match user ${signer} != ${gdAddress}`)
+    }
+  }
+
+  async getEnrollment(enrollmentIdentifier: string, customLogger = null): Promise<any> {
```
Comment (sirpy) [inline src/server/verification/processor/EnrollmentProcessor.js:181]: revert this refactoring

File: src/server/goodid/goodid-middleware.js [hunk 3]
```diff
@@ -110,6 +114,94 @@ export default function addGoodIDMiddleware(app: Router, utils) {
...
+          throw new Error(enrollmentNotFoundMessage)
+        }
+
+        const { auditTrailBase64 } = await enrollmentProcessor.getEnrollment(faceIdentifier, log)
+        const estimation = await utils.ageGenderCheck(auditTrailBase64)
+
+        if ((fvAgeCheck || 'none') !== 'none') {
+          // strict or approximate
+          estimation.age = await enrollmentProcessor.estimateAge(faceIdentifier, fvAgeCheck === 'strict', log)
+        }
```
Comment (sirpy) [inline src/server/goodid/goodid-middleware.js:183]: not required

**Review by sirpy — CHANGES_REQUESTED** [review]

File: src/server/verification/processor/EnrollmentProcessor.js (hunk 2 above)
Comment (johnsmith-gooddollar) [reply src/server/verification/processor/EnrollmentProcessor.js:181]: ok, but 
a) I will create a separate verification/utils/utils.js file
b) will move verifyIdentifier from verificationAPI there (as it will be also used from goodID middleware)
c) will keep normalizeIdentifiers, but move it also to utils.js as a simple function. Because this code part repeats in any FV endpoint, more that 5 times. And, as we agreed, if something repeats more than 2-3 times - we could refactor it by making a shared util

File: src/server/goodid/goodid-middleware.js (hunk 3 above)
Comment (johnsmith-gooddollar) [reply src/server/goodid/goodid-middleware.js:183]: ok

File: src/server/goodid/utils.js [hunk 4]
```diff
@@ -62,6 +63,15 @@ export class GoodIDUtils {
     return phoneUtil.getRegionCodeForNumber(number)
   }
 
+  async ageGenderCheck(imageBase64) {
+    const { FaceDetails } = await detectFaces(imageBase64)
+    const [{ AgeRange, Gender }] = FaceDetails
+    const { Value: gender } = Gender
+    const { Low: from, High: to } = AgeRange
+
+    return { gender, age: { from, to } }
```
[the remark this answers is not in the document]
Comment (johnsmith-gooddollar) [reply src/server/goodid/utils.js:72]: ok

File: src/server/goodid/goodid-middleware.js [hunk 5]
```diff
@@ -163,3 +258,43 @@ export default function addGoodIDMiddleware(app: Router, utils) {
...
+    const { FaceDetails } = await detectFaces(auditTrailBase64)
+    log.info({ FaceDetails })
+    await Promise.all([
+      // semaphore.enrollAge(FaceDetails[0].AgeRange),
+      // semaphore.enrollGender(FaceDetails[0].Gender.Value)
+    ])
+
+    res.json({ ok: 1 })
+  })
+)*/
```
Comment (johnsmith-gooddollar) [inline src/server/goodid/goodid-middleware.js:300]:
```suggestion
```

**Review by sirpy — APPROVED** [review]
